controllers/operatorpolicy_controller.go

```diff
@@ -495,6 +526,143 @@ func buildSubscription(
 	return subscription, nil
 }
 
+func (r *OperatorPolicyReconciler) handleCSV(ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	sub *operatorv1alpha1.Subscription,
+) (*operatorv1alpha1.ClusterServiceVersion, error) {
+	// case where subscription is nil
+	if sub == nil {
+		// need to remove all relatedobjects
+		err := r.removeObjsOfType(ctx, policy, clusterServiceVersionGVK)
+		if err != nil {
+			return nil, fmt.Errorf("error removing relatedObjects: %w", err)
+		}
+
+		err = r.updateStatus(ctx, policy, noDeploymentsCond, noExistingCSVObj)
+		if err != nil {
+			return nil, fmt.Errorf("error updating the status for Deployments: %w", err)
+		}
+
+		return nil, nil
+	}
+
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	// case where status has not been populated yet
+	if sub.Status.InstalledCSV == "" {
+		return nil, nil
+	}
+
+	// Get the CSV related to the object
+	foundCSV, err := r.DynamicWatcher.Get(watcher, clusterServiceVersionGVK, sub.Namespace,
+		sub.Status.InstalledCSV)
+	if err != nil {
+		return nil, err
+	}
+
+	// CSV has not yet been created by OLM
+	if foundCSV == nil {
+		err := r.updateStatus(ctx, policy,
+			missingWantedCond("ClusterServiceVersion"), missingCSVObj(sub.Name, sub.Namespace))
+		if err != nil {
+			return nil, fmt.Errorf("error updating the status for a missing ClusterServiceVersion: %w", err)
+		}
+
+		return nil, err
+	}
+
+	// Check CSV most recent condition
+	unstructured := foundCSV.UnstructuredContent()
+	var csv operatorv1alpha1.ClusterServiceVersion
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured, &csv)
+
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.updateStatus(ctx, policy, buildCSVCond(&csv), existingCSVObj(&csv))
+	if err != nil {
+		return &csv, fmt.Errorf("error updating the status for an existing ClusterServiceVersion: %w", err)
+	}
+
+	return &csv, nil
+}
+
+func (r *OperatorPolicyReconciler) handleDeployment(
+	ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	csv *operatorv1alpha1.ClusterServiceVersion,
+) error {
+	// case where csv is nil
+	if csv == nil {
+		// need to remove all relatedobjects
+		err := r.removeObjsOfType(ctx, policy, deploymentGVK)
+		if err != nil {
+			return fmt.Errorf("error removing relatedObjects: %w", err)
+		}
+
+		err = r.updateStatus(ctx, policy, noDeploymentsCond, noExistingDeploymentObj)
```

Comment:
Same comment as for CSVs: if `updateStatus` gets `noExistingDeploymentObj`, then the `removeObjsOfType` is unnecessary.

Reply:
I also used the wrong `Cond`, my bad